kedro-datasets/kedro_datasets_experimental/langchain/langchain_prompt_dataset.py

```diff
@@ -0,0 +1,315 @@
+    """
+
+    TEMPLATES = {
+        "PromptTemplate": "_create_prompt_template",
+        "ChatPromptTemplate": "_create_chat_prompt_template",
+    }
+
+    def __init__(  # noqa: PLR0913
+        self,
+        filepath: str,
+        template: str = "PromptTemplate",
+        dataset: dict[str, Any] | str | None = None,
+        credentials: dict[str, Any] | None = None,
+        fs_args: dict[str, Any] | None = None,
+        metadata: dict[str, Any] | None = None,
+        **kwargs: Any,
+    ):
+        """
+        Initialize the LangChain prompt dataset.
+
+        Args:
+            filepath: Path to the prompt file
+            template: Name of the LangChain template class ("PromptTemplate" or "ChatPromptTemplate")
+            dataset: Configuration for the underlying Kedro dataset
+            credentials: Credentials passed to the underlying dataset unless already defined
+            fs_args: Extra arguments passed to the filesystem, if supported
+            metadata: Arbitrary metadata
+            **kwargs: Additional arguments (ignored)
+        """
+        super().__init__()
+
+        self.metadata = metadata
+        self._filepath = get_filepath_str(Path(filepath), kwargs.get("protocol"))
+
+        try:
+            self._template_name = template
+            self._create_template_function = getattr(self, self.TEMPLATES[template])
+        except KeyError:
+            raise DatasetError(
+                f"Invalid template '{template}'. Must be one of: {list(self.TEMPLATES)}"
+            )
+
+        # Infer dataset type if not explicitly provided
+        dataset_config = self._build_dataset_config(dataset)
+
+        # Handle credentials
+        self._credentials = deepcopy(credentials or {})
+        self._fs_args = deepcopy(fs_args or {})
+
+        if self._credentials:
+            if CREDENTIALS_KEY in dataset_config:
+                self._logger.warning(
+                    "Top-level credentials will not propagate into the underlying dataset "
+                    "since credentials were explicitly defined in the dataset config."
+                )
+            else:
+                dataset_config[CREDENTIALS_KEY] = deepcopy(self._credentials)
+
+        if self._fs_args:
+            if "fs_args" in dataset_config:
+                self._logger.warning(
+                    "Top-level fs_args will not propagate into the underlying dataset "
+                    "since fs_args were explicitly defined in the dataset config."
+                )
+            else:
+                dataset_config["fs_args"] = deepcopy(self._fs_args)
+
+        try:
+            dataset_class, dataset_kwargs = parse_dataset_definition(dataset_config)
+            self._dataset = dataset_class(**dataset_kwargs)
+        except Exception as e:
+            raise DatasetError(f"Failed to create underlying dataset: {e}")
+
+    def _build_dataset_config(self, dataset: dict[str, Any] | str | None) -> dict[str, Any]:
```

Comment:
I wonder if it's better to specify strictly what the underlying datasets can be - just `TextDataset`, `YAMLDataset` and `JSONDataset` and error out if it isn't instead of inferring it. Unlike `PartitionedDataset` it's not like this can be any underlying dataset type